packages/widgetbook_generator/lib/src/code/widgetbook_instance.dart

```diff
@@ -30,12 +32,24 @@ class WidgetbookInstance extends InvokeExpression {
           (child) => child is TreeNode<UseCaseMetadata>,
         );
 
-    return isComponentNode
-        ? WidgetbookComponentInstance(
-            node: node as TreeNode<String>,
-          )
-        : WidgetbookFolderInstance(
-            node: node as TreeNode<String>,
-          );
+    if (node is! TreeNode<String>) {
+      throw new InvalidGenerationSourceError(
+        'Invalid state: node is not a ${TreeNode<String>}',
+        todo: 'This is an issue within widgetbook_generator itself. '
+            'Please open an issue at https://github.com/widgetbook/widgetbook/issues/new',
+      );
+    }
+
+    if (isComponentNode) {
+      return WidgetbookComponentInstance(node: node);
+    }
+
+    final name = node.data;
+    final categoryRegex = RegExp(r'^\[.*?\]$');
+    final isCategory = categoryRegex.hasMatch(name);
+
+    return isCategory
```

Comment:
In terms of readability, I prefer using `name.startsWith('[') && name.endsWith(']')` over using regex.